gui/DepthMapOptions.cxx

```diff
@@ -0,0 +1,216 @@
+ */
+
+#include "DepthMapOptions.h"
+#include "ui_DepthMapOptions.h"
+
+#include <qtUiState.h>
+#include <qtUiStateItem.h>
+#include <QWidgetAction>
+#include <QMenu>
+
+#include <vtkActor.h>
+
+#include "DepthMapFilterOptions.h"
+
+///////////////////////////////////////////////////////////////////////////////
+
+//BEGIN DepthMapOptionsPrivate declaration
+
+//-----------------------------------------------------------------------------
+class DepthMapOptionsPrivate
+{
+public:
+  Ui::DepthMapOptions UI;
+  qtUiState uiState;
+
+  DepthMapFilterOptions* filterOptions;
+
+  void setPopup(QToolButton* button, QWidget* widget);
+
+};
+
+//END DepthMapOptionsPrivate definition
+
+///////////////////////////////////////////////////////////////////////////////
+
+//BEGIN DepthMapOptionsPrivate implementation
+
+//-----------------------------------------------------------------------------
+void DepthMapOptionsPrivate::setPopup(QToolButton* button, QWidget* widget)
+{
+  auto const proxy = new QWidgetAction(button);
+  proxy->setDefaultWidget(widget);
+
+  auto const menu = new QMenu(button);
+  menu->addAction(proxy);
+
+  button->setMenu(menu);
+}
+
+//END DepthMapOptionsPrivate implementation
+
+///////////////////////////////////////////////////////////////////////////////
+
+//BEGIN DepthMapOptions
+
+QTE_IMPLEMENT_D_FUNC(DepthMapOptions)
+
+//-----------------------------------------------------------------------------
+DepthMapOptions::DepthMapOptions(QString const& settingsGroup,
+                                 QWidget* parent, Qt::WindowFlags flags) :
+  QWidget(parent, flags), d_ptr(new DepthMapOptionsPrivate)
+{
+  QTE_D();
+
+  // Set up UI
+  d->UI.setupUi(this);
+
+  // Set up option persistence
+  d->uiState.setCurrentGroup(settingsGroup);
+
+  d->uiState.restore();
+
+  d->filterOptions = new DepthMapFilterOptions(settingsGroup, this);
+  d->setPopup(d->UI.toolButtonFilters, d->filterOptions);
+  d->UI.toolButtonFilters->setEnabled(false);
+
+  // Connect signals/slots
+  connect(d->UI.radioPoints, SIGNAL(toggled(bool)),
+          this, SIGNAL(depthMapRepresentationChanged()));
+
+  connect(d->UI.radioSurfaces, SIGNAL(toggled(bool)),
+          this, SIGNAL(depthMapRepresentationChanged()));
+
+  connect(d->UI.checkBoxFilters, SIGNAL(toggled(bool)),
+          this, SLOT(showFiltersMenu(bool)));
+
+  connect(d->filterOptions, SIGNAL(filtersChanged()),
+          this, SIGNAL(depthMapThresholdsChanged()));
+}
+
+//-----------------------------------------------------------------------------
+DepthMapOptions::~DepthMapOptions()
+{
+  QTE_D();
+  d->uiState.save();
+}
+
+//-----------------------------------------------------------------------------
+void DepthMapOptions::showFiltersMenu(bool state)
+{
+  QTE_D();
+
+  d->UI.toolButtonFilters->setEnabled(state);
+}
+
+//-----------------------------------------------------------------------------
+void DepthMapOptions::enable()
```

Comment:
What's the purpose of this method? I don't see any code that would ever disable this, and AFAIK we don't really support changing data sets once data is loaded (for forcing the mode to points to be useful).